### PR TITLE
fix: Strategy heuristic test & fix

### PR DESCRIPTION
**Pull request by chloedia**
- Add heuristic tests with a few open source pdf files source: https://digitalcorpora.org/corpora/file-corpora/cc-main-2021-31-pdf-untruncated/
- fix strategy to include vectorized text as images

### CODE REVIEW

File: libs/megaparse/tests/pdf/test_detect_ocr.py [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+list_pdf = os.listdir("./tests/pdf/ocr")
+
 
-def test_strategy_all():
-    pdf = "./tests/pdf/sample_pdf.pdf"
+@pytest.mark.parametrize("pdf", list_pdf)
+def test_strategy_all(pdf):
     strategy = determine_strategy(
-        pdf, threshold_pages_ocr=0.2, threshold_image_page=0.3
+        f"./tests/pdf/ocr/{pdf}", threshold_pages_ocr=0.2, threshold_image_page=0.3
```
Comment (AmineDiro) [inline libs/megaparse/tests/pdf/test_detect_ocr.py:13]: Can we have also the inverse tests ? Native pdf documents should not be considered scanned

File: libs/megaparse/tests/test_parsers.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+@pytest.mark.parametrize("parser", PARSER_LIST)
+@pytest.mark.parametrize("extension", list(FileExtension))
+def test_sync_parser(parser, extension):
+    directory = "./tests/supported_docs"
+    file_path = next(
+        (
+            os.path.join(root, file)
+            for root, _, files in os.walk(directory)
+            for file in files
+            if file.endswith(extension.value)
```
Comment (AmineDiro) [inline libs/megaparse/tests/test_parsers.py:25]: We can move filter of parser supported extensions here

**Review by AmineDiro — COMMENTED** [review]
We need to be sure that we have the rights to publish these documents on github for everyone to see

**Comment by chloedia** [comment]
> We need to be sure that we have the rights to publish these documents on github for everyone to see

PDF Sources : https://digitalcorpora.org/corpora/file-corpora/cc-main-2021-31-pdf-untruncated/
Link to the term of use : https://commoncrawl.org/terms-of-use

Conclusion : It can be used for testing usage with limited docs inside the repo and we **must** cite the data source in the ReadMe 

@AmineDiro

File: libs/megaparse/tests/test_parsers.py (hunk 2 above)
Comment (chloedia) [reply libs/megaparse/tests/test_parsers.py:25]: I am not sure to understand

Comment (chloedia) [reply libs/megaparse/tests/test_parsers.py:25]: I also want to verify it throws an error when the specific parser does not support this extension, for example : Unstructured supports every possible extensions in MegaParse but doctr only supports PDF for the moment

**Pull request merged**